src/coreclr/jit/codegencommon.cpp

```diff
@@ -4176,59 +4220,50 @@ void CodeGen::genHomeSwiftStructParameters(bool handleStack)
             }
             else
             {
-                var_types loadType = TYP_UNDEF;
-                switch (seg.Size)
-                {
-                    case 1:
-                        loadType = TYP_UBYTE;
-                        break;
-                    case 2:
-                        loadType = TYP_USHORT;
-                        break;
-                    case 3:
-                    case 4:
-                        loadType = TYP_INT;
-                        break;
-                    case 5:
-                    case 6:
-                    case 7:
-                    case 8:
-                        loadType = TYP_LONG;
-                        break;
-                    default:
-                        assert(!"Unexpected segment size for struct parameter not passed implicitly by ref");
-                        continue;
-                }
+                // We can use REG_SCRATCH as a temporary register here as we ensured that during LSRA build.
+                genHomeStackSegment(lclNum, seg, REG_SCRATCH, nullptr);
+            }
+        }
+    }
+}
+#endif
 
-                int offset;
-                if (isFramePointerUsed())
-                {
-                    offset = -genCallerSPtoFPdelta();
-                }
-                else
-                {
-                    offset = -genCallerSPtoInitialSPdelta();
-                }
+/*-----------------------------------------------------------------------------
+ *
+ * Home the tail (stack) portion of a split parameter next to where the head (register) portion is homed.
+ *
+ * No-op on platforms where argument registers are already homed to form a contiguous space with incoming stack.
+ */
+void CodeGen::genHomeStackPartOfSplitParameter(regNumber initReg, bool* initRegStillZeroed)
+{
+#ifdef TARGET_RISCV64
+    unsigned lclNum = 0;
+    for (; lclNum < compiler->info.compArgsCount; lclNum++)
+    {
+        LclVarDsc* var = compiler->lvaGetDesc(lclNum);
+        if (!var->lvIsSplit)
+            continue;
```

Comment:
At some point I imagine we'll end up removing `LclVarDsc::lvIsSplit`, but this can stay for now.